src/modules/thresholdsig/threshold.md

```diff
@@ -106,39 +107,40 @@ verify_signature
 To produce a signature, each signer `i` acts as follows.
 
 1. Produces a nonce pair `(k_i, R_i)` and a commitment (hash) `C_i` to `R_i`.
-   This is done with `secp256k1_musig_multisig_generate_nonce`. Sends the
+   This is done with `secp256k1_musig_session_get_public_nonce`. Sends the
    commitment `C_i` to every other signer.
 2. Once at least `k` nonce commitments have been received from other signers,
-   signer `i` creates `n` `secp256k1_musig_signer_data` structures, one for
-   each signer including herself.
-   She initializes each with `secp256k1_musig_signer_data_initialize` which
+   signer `i` creates `n` `secp256k1_musig_session_signer_data` structures, one
+   for each signer including herself.
+   She initializes each with `secp256k1_thresholdsig_session_initialize` which
    takes a public key from all signers and a nonce commitment from present
    signers. If `k = n`, the public key should be `Y_i` from step 2 of the key
    setup; otherwise it should be `Z_i` from step 5 of key setup.
 3. She sends her actual public nonce `R_i` to every other signer.
-4. Upon receipt of another signer's public nonce, she calls `secp256k1_musig_set_nonce`
-   to update that signer's data structure. If the public nonce does not match
-   the precommitment, this function will fail and the signer will be considered
-   not to be present.
+4. Upon receipt of another signer's public nonce, she calls
+   `secp256k1_musig_set_nonce` to update that signer's data structure.
+   If the public nonce does not match the precommitment, this function will fail
+   and the signer will be considered not to be present.
 5. Once `k` valid public nonces have been received, she can produce a partial
-   signature using `secp256k1_musig_partial_sign`. She sends this partial signature
-   to someone (or everyone) for aggregation.
+   signature using `secp256k1_musig_partial_sign`. She sends this partial
+   signature to someone (or everyone) for aggregation.
    [TODO] Things will work with more than `k` signers; if people disagree on the
    set of signers this is a problem, so "just take the first `k`" is actually
    likely to fail in the case of `>k`.
 6. Some participant receives all the partial signatures and combines them using
-   `secp256k1_musig_combine_partial_sigs`. The output of this function is a
+   `secp256k1_musig_partial_sig_combine`. The output of this function is a
    complete signature. If the signature is invalid, at least one of the partial
    signatures was invalid. The culprit may be identified using the function
-   `secp256k1_musig_partial_sig_verify` on the individual partial signatures.
+   `secp256k1_thresholdsig_partial_sig_verify` on the individual partial
+   signatures.
 
 Signing Procedure Flowchart:
 ```
 Me:                           Others:
 
-multisig_generate_nonce C ->
-                           <- multisig_generate_nonce C
-signer_data_initialize
+musig_generate_nonce C    ->
```

Comment:
There's no `musig_generate_nonce`. This is now done in `session_initialize`. How about swapping `musig_generate_nonce` with `session_initialize` and replacing `musig_generate_nonce` with `musig_get_public_nonce`?